packages/dds/tree/src/feature-libraries/editable-tree/editableTreeContext.ts

```diff
@@ -0,0 +1,222 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import { assert } from "@fluidframework/common-utils";
+import { TransactionResult } from "../../checkout";
+import { Dependent, SimpleObservingDependent, InvalidationToken } from "../../dependency-tracking";
+import { IEditableForest, ITreeSubscriptionCursor, ITreeCursor, IForestSubscription, TreeNavigationResult } from "../../forest";
+import { lookupGlobalFieldSchema } from "../../schema-stored";
+import { ISharedTree } from "../../shared-tree";
+import { Delta, rootFieldKey, UpPath, Value } from "../../tree";
+import { Brand } from "../../util";
+import { Multiplicity } from "../modular-schema";
+import { NodePath, SequenceEditBuilder } from "../sequence-change-family";
+import { RootedTextCursor } from "../treeTextCursorLegacy";
+import { proxifyField, ProxyTarget, UnwrappedEditableField } from "./editableTree";
+import { ProxyTargetSequence } from "./editableTreeSequence";
+import { getFieldKind, pathToString } from "./utilities";
+
+/**
+ * A common context of a "forest" of EditableTrees.
+ * It handles group operations like transforming cursors into anchors for edits.
+ * TODO: add test coverage.
+ */
+export interface EditableTreeContext {
+    /**
+     * Gets a Javascript Proxy providing a JavaScript object like API for interacting with the tree.
+     *
+     * Use built-in JS functions to get more information about the data stored e.g.
+     * ```
+     * for (const key of Object.keys(context.root)) { ... }
+     * // OR
+     * if ("foo" in data) { ... }
+     * context.free();
+     * ```
+     */
+    readonly root: UnwrappedEditableField;
+
+    /**
+     * Call before editing.
+     *
+     * Note that after performing edits, EditableTrees for nodes that no longer exist are invalid to use.
+     * TODO: maybe add an API to check if a specific EditableTree still exists,
+     * and only make use other than that invalid.
+     */
+    prepareForEdit(): void;
+
+    /**
+     * Call to free resources.
+     * EditableTrees created in this context are invalid to use after this.
+     */
+    free(): void;
+
+    /**
+     * Register a handler function to be called after changes applied to the forest.
+     */
+    registerAfterHandler(afterHandler: EditableTreeContextHandler): void;
+}
+
+export type EditableTreeContextHandler = (this: EditableTreeContext) => void;
+
+export type ETreeNodePath = Brand<string, "editable-tree.NodePath">;
```

Comment:
Can you document this? What Seperators does it use? How does it handle global vs local field keys? Does it use escaping? How are indexes handled? Why does it exist?

Might be better to put it in its own file alongside the functions for working with it.